Update blog.md with clearer instructions (#3548)
open-telemetry/opentelemetry.io: archetypes/blog.md

## Top-level heading

> Top-level headings start at **level 2**. Your post should not include `# headings` but should instead only use `## headings`.

Reviewer: ```suggestion
Top-level headings start at **level 2**. This means, that your post should not include `# headings` for top-level headings but `## headings` instead.
```

The sentence _can_ lead to the assumption that we only accept level 2 headings, but of course they can use level3, level4, etc if they want to.

If you have an image stored at `content/en/{{ .File.Dir }}imagename.png`, you
can reference them like the following:

> `![Provide a good image description for improved accessibility](imagename.png)`

Reviewer: ```suggestion
![Provide a good image description for improved accessibility](imagename.png)
```

I disagree with this change, although this will show up as a broken image, the intend is that if someone runs the hugo new command they see how an image will be used.

Author: Maybe have both, so that users who may be unfamiliar with images in Markdown can see what it would look like?

Reviewer: > Maybe have both, so that users who may be unfamiliar with images in Markdown can see what it would look like?

Not sure if I follow. Users can see how it looks like by taking a look at the file & the preview? We can add a dummy image here to _unbreak_ it and make it clearer by that

```

Happy writing!

> **Note:** You can safely ignore the warning at the top of this page.

Reviewer: ```suggestion
**Note:** If you view this page with the GitHub file viewer, you can safely ignore the `Error in user YAML` at the top of this page.
```

> <img width="1024" alt="2023-11-15_10-32-38" src="https://github.com/davidgs/opentelemetry.io/assets/2071898/d7da3f54-64df-4120-b4f7-861134eb71d9">

Reviewer: ```suggestion
```
